commands/main.go

```diff
@@ -306,8 +306,8 @@ func getAPIAddress(req *cmds.Request) (string, error) {
 }
 
 func requiresDaemon(req *cmds.Request) bool {
-	for _, cmd := range rootSubcmdsLocal {
-		if req.Command == cmd {
+	for cmd := range rootSubcmdsLocal {
+		if len(req.Path) > 0 && req.Path[0] == cmd {
```

Comment:
The value of `req.Path[0]` here is going to be the first argument supplied to `go-filecoin`, for example:
- `go-filecoin daemon` req.Path[0] == "daemon"
- `go-filecoin version` req.Path[0] == "version"
- `go-filecoin chain ls` req.Path[0] == "chain"

The modification to the loop will cause `cmd` to be one of the below keys
https://github.com/filecoin-project/go-filecoin/blob/0ed37844d780810b6733680190fb21eccdecc991/commands/main.go#L154-L159

The length check on `req.Path` is to avoid panics when we don't pass arguments, is this all correct based on your understanding @deaswang?